src/main/python/e2e_experimental_settings.py

```diff
@@ -105,85 +105,83 @@ def run(self, indices):
-    #     'qa-data': '--idf-source qa-data',
-    #     'corpus-index': '--idf-source corpus-index'
-    # }))
-
-    # experiments.add_setting(Setting('punctuation', {
-    #     'keep': '--punctuation keep',
-    #     'remove': '--punctuation remove'
-    # }))
-
-    # experiments.add_setting(Setting('dash_words', {
-    #     'keep': '--dash-words keep',
-    #     'split': '--dash-words split'
-    # }))
-
-    experiments.add_setting(Setting('num_hits', {
-        '10':  '-hits 10',
-        '20':  '-hits 20',
-        '50':  '-hits 50',
-        '100': '-hits 100',
-        '200': '-hits 500'
-    }))
-
-    experiments.add_setting(Setting('top_k', {
-        '10': '-k 10',
-        '50': '-k 50',
-        '100': '-k 100',
-        '500': '-k 500',
-        '1000': '-k 1000',
-        '5000': '-k 5000'
-    }))
-
-    experiments.list_settings()
-
-    if args.run:
-        experiments.run(args.run)
-
-    if args.run_all:
-        experiments.run_all()
+                self._run_eval(params)
+
+        def run_all(self):
+            """
+            runs all experiments
+            """
+            self.run([ci for ci in range(len(self.combinations))])
+
+        def list_settings(self):
+            """
+            lists all settings
+            """
+            for c in enumerate(self.combinations):
+                print(c)
+            print("--run X Y Z to run combinations number X Y Z")
+
+    if __name__ == "__main__":
+        ap = argparse.ArgumentParser(description="Lists exerimental settings and runs experiments")
+        ap.add_argument("--list", help="lists all available experimental settings combinations",
+                        action="store_true")
+        ap.add_argument("--run", help='runs experimenal setting combination NUMBER(s)',
+                        nargs="+", type=int)
+        ap.add_argument("--run-all", action="store_true")
+        ap.add_argument('qa_data_xml', help="path to the QA dataset XML file")
+        ap.add_argument('qrel_file', help="the qrel file e.g. ../data/TrecQA/raw-dev.qrel")
+        ap.add_argument("index_path", help="required for some combination of experiments")
+        ap.add_argument('word_embeddings_file', help="the word embeddings file")
+        ap.add_argument('model_file', help="the model to be used for QA similarity")
+        # ap.add_argument("--idf-source", choices=["qa-data", "corpus-index"], default="corpus-index")
+        # ap.add_argument("--punctuation", choices=["keep", "remove"], default="keep")
+        # ap.add_argument("--dash-words", choices=["keep", "split"], default="keep")
+        # ap.add_argument("models_root", help="path to folder containing models")
+
+        args = ap.parse_args()
+
+        experiments = Experiments(args.qa_data_xml, args.qrel_file, args.word_embeddings_file,
+                                  args.index_path, args.model_file)
+
+        # experiments.add_setting(Setting('idf_source', {
```

Comment:
It is best if we can control all the experiment settings from the command line arguments rather than comment/uncommenting code.